### DISCUSSION TITLE
Loadable should be in a loading state whenever an atom throws a promise

### DISCUSSION THREAD

**Discussion by Vakrim** [pmndrs/jotai#2598] [Bug report]
Currently, you can throw a promise from an atom to trigger suspense in React. This behavior is used in `atomWithInfiniteQuery` and `atomWithSuspenseInfiniteQuery` from `jotai-tanstack-query`.

However, when you wrap this type of atom within `loadable()`, it returns `{state: 'hasError', error: promise}` rather than the expected `{state: 'loading'}`.

**Comment by dai-shi** [comment]
> you can throw a promise from an atom to trigger suspense in React.

It won't work in the future versions of React, so I'd suggest to fix `jotai-tanstack-query`. @kalijonn

**Comment by kalijonn** [comment]
@Vakrim I'm unable to reproduce this. I've wrapped both InfiniteQuery and SuspenseInfiniteQuery and see state as 'loading'. Can you create a minimal reproduction?

@dai-shi We're not throwing from within the atom. We return a promise and useAtom throws it to suspend. This is for suspense queries.

Lodable utility is not necessary for jotai-tanstack-query because we're separated the suspending and normal queries. If you don't want the component to suspend, you can use the normal atomWith\*Query. If you want the component to suspend, you can use atomWithSuspense\*Query.

You can technically still use loadable on top of atomWithSuspense\*Query which returns {state: 'loading'}. Not sure how this is useful over just using atomWith\*Query

**Reply by dai-shi** [reply]
> We're not throwing from within the atom. We return a promise and useAtom throws it to suspend. This is for suspense queries.

Yeah, I get it.

> Lodable utility is not necessary for jotai-tanstack-query because we're separated the suspending and normal queries.

That's true usage wise, but I'm curious if there's a problem with the combination with ` atomWithSuspense*Query` and `loadable`. Not a priority though.

**Comment by Vakrim** [comment]
I tried to reproduce it, but I wasn't successful. I think I got confused somewhere in my code previously. Sorry for the trouble
